src/material-experimental/mdc-slider/slider.ts

```diff
@@ -0,0 +1,517 @@
+        },
+    setTrackStyleProperty:
+        (propertyName, value) => {
+          this._track.nativeElement.style.setProperty(propertyName, value);
+        },
+    setMarkerValue:
+        () => {
+          // Mark the component for check as the thumb label needs to be re-rendered.
+          this._changeDetectorRef.markForCheck();
+        },
+    setTrackMarkers:
+        (step, max, min) => {
+          this._trackMarker.nativeElement.style.setProperty(
+              'background', this._getTrackMarkersBackground(min, max, step));
+        },
+    isRTL: () => this._dir && this._dir.value === 'rtl',
+  };
+
+  /** Instance of the MDC slider foundation for this slider. */
+  private _foundation = new MDCSliderFoundation(this._sliderAdapter);
+
+  /** Whether the MDC foundation has been initialized. */
+  private _isInitialized = false;
+
+  /** Function that notifies the control value accessor about a value change. */
+  private _controlValueAccessorChangeFn: (value: number) => void = () => {};
+
+  /** Subscription to the Directionality change EventEmitter. */
+  private _dirChangeSubscription = Subscription.EMPTY;
+
+  /** Function that marks the slider as touched. Registered via "registerOnTouch". */
+  _markAsTouched: () => any = () => {};
+
+  @ViewChild('thumbContainer', {static: false}) _thumbContainer: ElementRef<HTMLElement>;
+  @ViewChild('track', {static: false}) _track: ElementRef<HTMLElement>;
+  @ViewChild('pinValueMarker', {static: false}) _pinValueMarker: ElementRef<HTMLElement>;
+  @ViewChild('trackMarker', {static: false}) _trackMarker: ElementRef<HTMLElement>;
+
+  constructor(
+      private _elementRef: ElementRef<HTMLElement>, private _changeDetectorRef: ChangeDetectorRef,
+      private _ngZone: NgZone, @Optional() private _dir: Directionality,
+      @Attribute('tabindex') tabIndex: string,
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    this.tabIndex = parseInt(tabIndex) || 0;
+
+    if (this._dir) {
+      this._dirChangeSubscription = this._dir.change.subscribe(() => {
+        // In case the directionality changes, we need to refresh the rendered MDC slider.
+        // Note that we need to wait until the page actually updated as otherwise the
+        // client rectangle wouldn't reflect the new directionality.
+        // TODO(devversion): ideally the MDC slider would just compute dimensions similarly
+        // to the standard Material slider on "mouseenter".
+        this._ngZone.runOutsideAngular(() => setTimeout(() => this._foundation.layout()));
+      });
+    }
+  }
+
+  ngAfterViewInit() {
+    this._isInitialized = true;
+    this._foundation.init();
+
+    // The standard Angular Material slider is always using discrete values. We always
+    // want to enable discrete values and support ticks, but want to still provide
+    // non-discrete slider visual looks if thumb label is disabled.
+    // TODO(devversion): check if we can get a public API for this.
```

Comment:
Can you file an issue on their repo?

Reply:
Done. https://github.com/material-components/material-components-web/issues/5020